server/src/main/java/org/opensearch/index/remote/RemoteIndexPath.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.remote;
+
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.common.blobstore.BlobPath;
+import org.opensearch.core.xcontent.ToXContentFragment;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.index.remote.RemoteStoreEnums.DataCategory;
+import org.opensearch.index.remote.RemoteStoreEnums.DataType;
+import org.opensearch.index.remote.RemoteStoreEnums.PathHashAlgorithm;
+import org.opensearch.index.remote.RemoteStoreEnums.PathType;
+import org.opensearch.index.remote.RemoteStorePathStrategy.PathInput;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static org.opensearch.index.remote.RemoteStoreEnums.DataCategory.SEGMENTS;
+import static org.opensearch.index.remote.RemoteStoreEnums.DataCategory.TRANSLOG;
+import static org.opensearch.index.remote.RemoteStoreEnums.DataType.DATA;
+import static org.opensearch.index.remote.RemoteStoreEnums.DataType.METADATA;
+import static org.opensearch.index.remote.RemoteStorePathStrategy.isCompatible;
+
+/**
+ * Remote index path information.
+ *
+ * @opensearch.internal
+ */
+@ExperimentalApi
+public class RemoteIndexPath implements ToXContentFragment {
+
+    public static final Map<DataCategory, List<DataType>> TRANSLOG_PATH = Map.of(TRANSLOG, List.of(DATA, METADATA));
+    public static final Map<DataCategory, List<DataType>> SEGMENT_PATH = Map.of(SEGMENTS, List.of(DataType.values()));
+    public static final Map<DataCategory, List<DataType>> COMBINED_PATH;
+
+    static {
+        Map<DataCategory, List<DataType>> combinedPath = new HashMap<>();
+        combinedPath.putAll(TRANSLOG_PATH);
+        combinedPath.putAll(SEGMENT_PATH);
+        COMBINED_PATH = Collections.unmodifiableMap(combinedPath);
+    }
+    private static final String DEFAULT_VERSION = "1";
+    public static final String DIR = "remote-index-path";
+    public static final String FILE_NAME_FORMAT = "remote_path_%s";
```

Comment:
It should have codec version that dictates how the file needs to be parsed. If we change the file format, like SMILE or anything else we might not be even able to read the file.

Reply:
Let me take this up in a follow up PR that I will be raising. I will be adding some details about the version, index metadata nitty-gritty in the filename.This PR has UTs that validates the content and ITs that assert the presence of the file, but let me also try to validate that the encoding & contents of the file are as expected.

Reply:
https://github.com/opensearch-project/OpenSearch/pull/13386